doc/model.rst

```diff
@@ -0,0 +1,719 @@
+
+Assuming an initialized ``Model`` named `smartsim_model` exists, we add a TensorFlow model using
+the ``Model.add_script()`` function:
+
+.. code-block:: python
+
+    smartsim_model.add_script(name="example_script", script=torch_script_str, device="GPU", devices_per_node=2, first_device=0)
+
+In the above ``smartsim_model.add_script()`` code snippet, we offer the following arguments:
+
+-  name ("example_script"): key to store script under.
+-  script (torch_script_str): TorchScript code.
+-  device ("CPU"): device for script execution.
+-  devices_per_node (2): Use two GPUs per node.
+-  first_device (0): Start with 0 index GPU.
+
+When the model is started via ``Experiment.start()``, the TorchScript will be loaded to the
+orchestrator that is launched prior to the start of the model.
+
+=========================
+Data Collision Prevention
+=========================
+--------
+Overview
+--------
+SmartSim and SmartRedis can avoid key collision by prepending program-unique
+prefixes to Model workloads launched through SmartSim. For example, if you were
+to have two applications feeding data to a single database, who produced keys
+of the same name, upon requesting this information there would be a key collision
+since there is no yet uniqueness between the same tensor names. By enabling key
+prefixing on a Model, SmartSim will append the model name to each key produced
+by the application and sent to the database as such: `model_name.tensor_name`.
+
+This is done simply through functions offered by the Model API:
+
+* Model.register_incoming_entity(incoming_entity)
+* Model.enable_key_prefixing()
+* Model.disable_key_prefixing()
+* Model.query_key_prefixing()
+
+-------
+Example
+-------
+We provide a producer/consumer example that demonstrates
+two producer models, with key prefixing enabled, that
+send tensor of the same name to a standard database. A
+consumer Model requests both tensors and displays the
+information using a SmartRedis logger to demonstrates
+successful use of Model key prefixing.
+
+During the example we will be creating four different files:
+
+1. producer_1.py : a Model producer application
+2. producer_2.py : a Model producer application
+3. consumer.py : a Model consumer application
+4. experiment.py : the Experiment driver script
+
+Producer_1 Application
+======================
+Begin by importing the necessary modules and initializing a SmartRedis
+Client:
+
+.. code-block:: python
+
+    from smartredis import Client
+    from smartredis import *
+    import numpy as np
+
+    # Initialize a Client
+    client = Client(cluster=False)
+
+Next, create a NumPy array to place in the database with the key name `tensor`:
+
+.. code-block:: python
+
+    # Create NumPy array
+    array = np.array([5, 6, 7, 8])
+    # Use SmartRedis client to place tensor in single sharded db
+    client.put_tensor("tensor", array)
+
+With key prefixing enabled, the tensor stored will be under key
+`producer_1.tensor`.
```

Comment:
users trip up on this so I think we should call it out.  We should talk about any necessary steps for a ``Model`` to retrieve its own data when key prefixing is on.